[DataPipe] Fixing IterDataPipe docstrings

NivekT opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72476
* __->__ #72475

To render the change in documentation, please pull down this PR and build the doc in `TorchData`.

Differential Revision: [D34078064](https://our.internmc.facebook.com/intern/diff/D34078064)

NivekT committed:
[DataPipe] Fixing IterDataPipe docstrings

[ghstack-poisoned]

ejguan commented on torch/utils/data/datapipes/iter/combining.py:
```diff
@@ -73,11 +72,10 @@ def __new__(cls, datapipe: IterDataPipe, num_instances: int, buffer_size: int =
 
 
 class _ForkerIterDataPipe(IterDataPipe):
```
Just want to confirm this is a private class and not shown in our doc even though we have doc strings.

ejguan commented on torch/utils/data/datapipes/iter/fileopener.py:
```diff
@@ -6,17 +6,15 @@
         datapipe: Iterable datapipe that provides pathnames
         mode: An optional string that specifies the mode in which
-            the file is opened by `open()`. It defaults to 'b' which
+            the file is opened by ``open()``. It defaults to ``b`` which
```
b and t her should be string. (Maybe: ``'b'`` and ``'t'``)

ejguan reviewed (approved):
Thank you so much. LGTM

NivekT committed:
Update on "[DataPipe] Fixing IterDataPipe docstrings"


To render the change in documentation, please pull down this PR and build the doc in `TorchData`.

[ghstack-poisoned]

NivekT commented:
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34078064).
